blocks/secure-assets/secure-assets.js

```diff
@@ -0,0 +1,152 @@
+// eslint-disable-next-line import/no-unresolved
+import { SignJWT } from 'https://cdnjs.cloudflare.com/ajax/libs/jose/5.2.3/jwt/sign.js';
+
+const securedImages = [];
+let placeholderImg;
+
+/**
+ * to check if given src is a DM OpenAPI URL
+ */
+function isDMOpenAPIUrl(src) {
+  return /^(https?:\/\/(.*)\/adobe\/assets\/urn:aaid:aem:(.*))/gm.test(src);
+}
+
+/**
+ * to check if given src is a secure asset
+ */
+async function isSecureAsset(src) {
```

Comment:
Can we skip this check and by default assume all assets in the `secure` block to be secured assets?

Reply:
Given the extended functionality, users have the flexibility to set section metadata, enabling this block to scan the entire page instead of just the block's scope. Secure assets may not always be confined to a specific block and can be present anywhere in the page DOM. For example, a homepage with exclusive section deals can contain both public and secure assets. Instead of reloading all images after user login, the block will refresh only the secure images, leaving public images untouched.

<img width="699" alt="image" src="https://github.com/user-attachments/assets/8d345b3c-1e3a-48d7-9edb-1d27d8823caa">

Reply:
maybe you can reuse the `onerror` event handler for the `img` tag to show placeholder in case of error. And then later, whenever the token is set, you restore the original image src with the token. Something like below

```
<img src="invalid-image-url.jpg" data-original-src="invalid-image-url.jpg" onerror="handleImageError(this)" alt="Image">
<img src="another-invalid-image.jpg" data-original-src="another-invalid-image.jpg" onerror="handleImageError(this)" alt="Another Image">
```

```
<script>
  // Function to handle the image error and set the placeholder
  function handleImageError(img) {
    img.setAttribute('data-original-src', img.src);  // Store original src
    img.onerror = null;  // Prevent infinite loop in case placeholder also fails
    img.src = 'placeholder.jpg';  // Set placeholder image
  }

  // Function to restore the original src for an image
  function restoreOriginalImage(img) {
    const originalSrc = img.getAttribute('data-original-src');
    if (originalSrc) {
      img.src = originalSrc;  // Restore original image source
    }
  }

  // Event listener for "token-available"
  document.addEventListener('token-available', function() {
    // Select all img elements that have the placeholder image
    const placeholderImg = 'placeholder.jpg';  // Define placeholder src
    const images = document.querySelectorAll('img');

    images.forEach(img => {
      if (img.src.includes(placeholderImg)) {  // Check if img has the placeholder
        restoreOriginalImage(img);  // Restore the original image source
      }
    });
  });
</script>

```

Reply:
I have incorporated your feedback partially in best possible fit for now as not using the `onError` functionality for `<img>` tag was intentional because of following reason : 
- the conversion of `<a>` tag to `<img>` is expected to be handled at scripts.js level likewise [External Images](https://github.com/hlxsites/franklin-assets-selector/blob/ext-images/EXTERNAL_IMAGES.md) and by the time flow gets on to secure asset block, the on error would have no effect unless we explicitly hard refresh all images.
- We need to consider the usecase where an author could have embedded secure assets along side public assets. So, IMO its essential to identify the secure images first similar to how I've tried in here by using HEAD request and then refresh/restore just the secure one's once token is available to consume.

So, making a HEAD request to identify public v/s secure images is the primary logic to avoid network overload is what the intention originally is. Same I have documented in README to for our end users as well.

Reply:
With the HEAD request, we'd end up making to network calls, first one to check if the image is secure and second to actually render the image. This seems be unacceptable if the image constituted the LCP as it may take a while to load. Can we setup a test page to figure that out?
If we do want to do via HEAD request, then I'd suggest to limit the support to the secure block only instead of full page or section and provide guidance in the doc to not use this block at places where it may end up impacting the page LHS.

Reply:
> then I'd suggest to limit the support to the secure block only instead of full page or section and provide guidance in the doc to not use this block at places where it may end up impacting the page LHS.

Thanks @sdmcraft for the suggestion. I have limited the usecase to block section instead of full page and have also added documentation in README to let our end user know that it may impact LCP if not used efficiently. Here is the doc ref - https://github.com/adobe-rnd/aem-assets-plugin/blob/issue-3-secure-asset/blocks/secure-assets/README.md#how-to-maintain-a-page-score-of-100-on-lhs-with-secure-assets-on-the-page